include/rcutils/filesystem.h

```diff
@@ -243,6 +244,75 @@ RCUTILS_PUBLIC
 size_t
 rcutils_get_file_size(const char * file_path);
 
+/// An iterator used for enumerating directory contents
+typedef struct rcutils_dir_iter_t
+{
+  /// The name of the enumerated file or directory
+  const char * entry_name;
+  /// The allocator used internally by iteration functions
+  rcutils_allocator_t allocator;
```

Comment:
This could be moved to the internal `rcutils_dir_iter_state_t` struct.

Reply:
I have no strong opinion about this.